cypress/CYPRESS.md

```diff
@@ -8,16 +8,106 @@ SEE: [README](../README.md) for instructions on setting up and running tests
+   - Avoid tests that require execution in a specific order.
+
+2. **Environment**:
+
+   - Tests should be executable in any environment without depending on the specific state of that environment.
+   - Design tests to be environment-agnostic, ensuring they can run in development and production environments.
+
+3. **Resource Management**:
+   - Each test should create all necessary resources for its execution and clean them up after completion.
+   - This includes data creation, user logins, and any other setup or teardown tasks.
+   - Leveraging the `before` and `after` hooks in Cypress can help manage resource creation and cleanup.
+
+### Test Implementation
+
+1. **Element Selection**:
+
+   - Prefer using data attributes (like `data-cy`) for selecting elements.
+   - Avoid using CSS classes or IDs for element selection, as they can change frequently and are not unique.
+
+2. **Asynchronous Operations**:
+
+   - Properly handle the asynchronous nature of web applications, using Cypress's automatic waiting features or custom waits when necessary.
+   - Avoid to use `cy.wait` as much as possible - mainly with hardcoded wait values, prefer to use `cy.get` with assertions.
+
+   **Example**:
+
+   ```javascript
+   describe('User Profile Test', () => {
+     it('loads user profiles asynchronously', () => {
+       // Visit the user profile page
+       cy.visit('/profiles');
+
+       // Interact with an element that triggers an asynchronous operation
+       cy.get('button.load-profiles').click();
+
+       // Cypress automatically waits for the element to appear in the DOM
+       cy.get('.profile-list').should('be.visible');
+
+       // Assert on the elements loaded as a result of the asynchronous operation
+       cy.get('.profile-list .profile-item').should('have.length', 5);
+
+       // You can also wait for specific text to appear
+       cy.contains('.profile-list', 'John Doe');
+
+       // Asserting on elements that are expected to update
+       cy.get('.profile-status').should('contain', 'Profiles Loaded');
+     });
+   });
+   ```
+
+3. **Test Data**:
+   - Utilize dynamic data creation methods to avoid conflicts with existing data.
+
+### Best Practices
+
+1. **Descriptive Test Names and Comments**:
+
+   - Write clear, descriptive names for test cases and suites.
+   - Add comments to explain complex logic or important test steps.
+
+2. **Security and Sensitive Data Handling**:
+
+   - Be cautious with sensitive data in tests, using mock or anonymized data where possible.
+
+3. **Performance Considerations**:
+   - Keep tests efficient to minimize impact on the development and deployment process.
+
+### Documentation and Collaboration
+
+1. **Documentation of Custom Commands and Utilities**:
+
+   - Document any Cypress [custom commands](./support/core-commands.ts) or utilities developed for the project.
+   - Include usage examples and explanations of the purpose of each command.
+
+2. **Team Collaboration**:
```

Comment:
Should we share a Slack channel for further conversations or related topics here?